### PR TITLE
improve DHT bootstrap performance

### PR DESCRIPTION
**Pull request by arvidn**
instead of restarting the whole DHT when changing external IP, just rebuild the routing table and change the node IDs

### CODE REVIEW

**Comment by ssiloti** [comment]
lgtm

File: src/kademlia/node.cpp [hunk 1]
```diff
@@ -115,8 +115,26 @@ node::node(udp_socket_interface* sock
...
+	// it's possible that our external address hasn't actually changed. If our
+	// current ID is still valid, don't do anything.
+	if (verify_id(m_id, m_observer->external_address()))
+		return;
+
+#ifndef TORRENT_DISABLE_LOGGING
+	if (m_observer) m_observer->log(dht_logger::node
+		, "updating node ID (because external IP address changed)");
+#endif
+
```
Comment (arvidn) [inline src/kademlia/node.cpp]: oops.. I think I forgot to set: m_id = generate_id(...); here :/

Comment (arvidn) [inline src/kademlia/node.cpp]: yeah, doing that made it perform a lot better..

**Comment by arvidn** [comment]
I should add a unit test for this new function

**Pull request merged**